packages/ui/src/shapes_ui.rs

```diff
@@ -0,0 +1,339 @@
+use crate::shapes::{Color, Geometry, Shape, Style};
+use crate::shapes_doc::{Document, ShapeId};
+use dioxus::prelude::*;
+use std::rc::Rc;
+
+const USE_MESSAGE_BOX: bool = false;
+
+const SHAPES_UI_CSS: Asset = asset!("/assets/styling/shapes_ui.css");
+
+enum TrackerNext {
+    Continue,
+    Done,
+}
+
+trait Tracker {
+    fn track_mouse_move(&self, evt: &MouseEvent) -> TrackerNext;
+    fn track_mouse_up(&self, evt: &MouseEvent);
+}
+
+static CANVAS_TRACKER: GlobalSignal<Option<Rc<dyn Tracker>>> = Global::new(|| None);
+
+static MESSAGE_BOX: GlobalSignal<String> = Global::new(|| "".to_string());
+
+struct AppState {
+    fill_color: Color,
+}
+
+impl AppState {
+    fn default() -> Self {
+        Self {
+            fill_color: Color::Red,
+        }
+    }
+
+    fn fill_color(&self) -> Color {
+        self.fill_color.clone()
+    }
+
+    fn get_fill_color_and_advance(&mut self) -> Color {
+        let old_color = self.fill_color.clone();
+        self.fill_color.advance();
+        old_color
+    }
+}
+
+static APP_STATE: GlobalSignal<AppState> = Global::new(|| AppState::default());
+
+#[component]
+pub fn SvgCanvasDiv() -> Element {
+    let opt_mouse_move_tracker = CANVAS_TRACKER();
+    let opt_mouse_up_tracker = opt_mouse_move_tracker.clone();
+
+    let mouse_move_handler = move |evt: MouseEvent| {
+        if let Some(tracker) = &opt_mouse_move_tracker {
+            evt.stop_propagation();
+            evt.prevent_default();
+            match tracker.track_mouse_move(&evt) {
+                TrackerNext::Continue => {}
+                TrackerNext::Done => {
+                    *CANVAS_TRACKER.write() = None;
+                }
+            };
+        }
+    };
+
+    let mouse_up_handler = move |evt: MouseEvent| {
+        if let Some(tracker) = &opt_mouse_up_tracker {
+            evt.stop_propagation();
+            evt.prevent_default();
+            tracker.track_mouse_up(&evt);
+            *CANVAS_TRACKER.write() = None;
+        }
+    };
+
+    rsx! {
+        div {
+            id: "svg_canvas_div",
+            onmousemove: mouse_move_handler,
+            onmouseup: mouse_up_handler,
+            SvgCanvas{}
+            if USE_MESSAGE_BOX {
+                MessageBox{}
+            }
+        }
+    }
+}
+
+#[component]
+fn MessageBox() -> Element {
+    if USE_MESSAGE_BOX {
+        let text: String = MESSAGE_BOX.read().clone();
+        rsx! {
+        div {
+            position: "absolute",
+            top: "0px",
+            left: "0px",
+            height: "20px",
+            right: "0px",
+            background_color: "blue",
```

Comment:
size=0x20?  What does this mean?  Don't you think something odd like this deserves a comment?